server/src/main/java/org/elasticsearch/http/AbstractHttpServerTransport.java

```diff
@@ -229,18 +235,40 @@ protected void doStop() {
                 }
             }
         }
-        try {
-            refCounted.decRef();
-            CloseableChannel.closeChannels(new ArrayList<>(httpChannels), true);
-        } catch (Exception e) {
-            logger.warn("unexpected exception while closing http channels", e);
+
+        gracefullyCloseConnections();
+        refCounted.decRef();
+
+        boolean closed = false;
+        if (shutdownGracePeriodMillis > 0) {
+            try {
+                FutureUtils.get(allClientsClosedListener, shutdownGracePeriodMillis, TimeUnit.MILLISECONDS);
+                closed = true;
+            } catch (ElasticsearchTimeoutException t) {
+                logger.trace(
```

Comment:
Should this be a warning? I would think we would want more visibility into whether this grace period was exhausted and connections are being closed forcefully.

Reply:
Changed to warning.